docs/src/modules.md

```diff
@@ -34,33 +34,43 @@ folder, using any name which is not on the list above.
 
 ## Module template
 
-All modules should have an enable option created using `mkEnableTarget`. This is
-similar to
-[`mkEnableOption`](https://nix-community.github.io/docnix/reference/lib/options/lib-options-mkenableoption/)
-from the standard library, however it integrates with
-[`stylix.enable`](./options/nixos.md#stylixenable) and
-[`stylix.autoEnable`](./options/nixos.md#stylixautoenable) and generates more
-specific documentation.
+Modules should be created using the `mkTarget` function whenever possible (see
+the `/stylix/mk-target.nix` documentation for more details):
 
-A general format for modules is shown below.
+```nix
+{ config, lib, mkTarget ... }:
+mkTarget {
+  name = "example";
+  humanName = "Example Target";
+
+  configElements =
+    { colors }:
+    {
+      programs.example.theme.background = colors.base00;
+    };
+}
+```
+
+When the `mkTarget` function cannot be used, modules must manually replicate its
+safeguarding functionality:
 
 ```nix
 { config, lib, ... }:
 {
-  options.stylix.targets.«name».enable =
-    config.lib.stylix.mkEnableTarget "«human readable name»" true;
+  options.stylix.targets.example.enable =
+    config.lib.stylix.mkEnableTarget "Example Target" true;
 
   config =
-    lib.mkIf (config.stylix.enable && config.stylix.targets.«name».enable)
+    lib.mkIf (config.stylix.enable && config.stylix.targets.example.enable)
       {
-        programs.«name».backgroundColor = config.lib.stylix.colors.base00;
+        programs.example.theme.background = config.lib.stylix.colors.base00;
       };
 }
```

Comment:
IMO it's clearer with the «unicode» arrows indicating placeholder values, I don't see any reason to change this example in this PR...

```suggestion
  options.stylix.targets.«name».enable =
    config.lib.stylix.mkEnableTarget "«human readable name»" true;

  config =
    lib.mkIf (config.stylix.enable && config.stylix.targets.«name».enable)
      {
        programs.«name».backgroundColor = config.lib.stylix.colors.base00;
      };
}
```

If anything, maybe the `mkTarget` example above should use the same placeholder pattern?